src/arrayMethodJoin.js

```diff
@@ -4,9 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        joinedString += '';
+      } else {
+        joinedString += this[i];
+      }
```

Comment:
try to rewrite this part a bit, I believe you could write a little less code here

Reply:
agree, try to rewrite it